spec/system/search_game_rooms_spec.rb

```diff
@@ -0,0 +1,61 @@
+require 'rails_helper'
+
+RSpec.describe 'SearchGameRooms', type: :system do
+  let(:user) { build(:user, points: 100) }
+  let(:track) { create(:track) }
+  let(:room) { create(:room) }
```

Comment:
### Note 1
Some suggestions 😃 
1. We can use the `let!` so that the room will be created before every example
2. We can already associate the track to the room upon creation. By updating the `test/factories/rooms.rb` and adding the following
```
name { 'animal' }
  trait :tracks do
    tracks { [association(:tracks)] }
  end
```
3. Then, omit the creation of room and tracks in the `before` block. Refer to Note 2.
```suggestion
  let!(:room) { create(:room, tracks: [track]) }
```